### DISCUSSION TITLE
Per unit envers config?

### DISCUSSION THREAD

**Discussion by Traivor** [quarkusio/quarkus#27971] [Q&A] [answered]
It doesn't appear that the hibernate-envers plugin supports configuration on a persistence unit basis. Am I reading things right? Are there any plans to support this?

Alternatively, would passing envers properties via the new quarkus.hibernate-orm."unit-name".unsupported-properties property work?

**Comment by gsmet** [comment]
@Traivor AFAICS, you can enable Envers per persistence unit. What exactly are you trying to do and can't?

**Comment by Traivor** [comment]
@gsmet this does not appear to be possible:

quarkus.hibernate-envers.db1.store-data-at-delete=true
quarkus.hibernate-envers.db1.audit-table-suffix=_aud
quarkus.hibernate-envers.db1.revision-field-name=rev
quarkus.hibernate-envers.db1.revision-type-field-name=revtype
quarkus.hibernate-envers.db1.audit-strategy=org.hibernate.envers.strategy.ValidityAuditStrategy
quarkus.hibernate-envers.db1.audit-strategy-validity-end-rev-field-name=revend
quarkus.hibernate-envers.db1.audit-strategy-validity-store-revend-timestamp=true
quarkus.hibernate-envers.db1.audit-strategy-validity-revend-timestamp-field-name=revend_tstmp

quarkus.hibernate-envers.db2.audit-table-suffix=_aud
quarkus.hibernate-envers.db2.revision-field-name=revision_id
quarkus.hibernate-envers.db2.revision-type-field-name=revtype
quarkus.hibernate-envers.db2.do-not-audit-optimistic-locking-field=false
quarkus.hibernate-envers.db2.store-data-at-delete=true

As far as I can tell you can only define envers properties at a global level, which isn't going to work very well if you have multiple persistence units which do envers differently from each other.

**Reply by gsmet** [reply]
Ah yes, you're right. I think this work predates my multiple persistence unit work. Let me create an issue for that.

**Reply by gsmet** [reply]
I created https://github.com/quarkusio/quarkus/issues/28133. Thanks for taking the time to report this and provide an example.

**Reply by codespearhead** [reply, marked as answer]
@Traivor The issue was discussed at https://github.com/quarkusio/quarkus/issues/28133 and addressed by PR https://github.com/quarkusio/quarkus/pull/28927 on `2022-10-31T10:20:22Z`.

Can you [mark this discussion as answered](https://docs.github.com/en/discussions/managing-discussions-for-your-community/moderating-discussions#marking-a-comment-as-an-answer) so it doesn't show up in the discussion tab as unanswered?

**Reply by Traivor** [reply]
> @Traivor The issue was discussed at #28133 and addressed by PR #28927 on `2022-10-31T10:20:22Z`.
> 
> Can you [mark this discussion as answered](https://docs.github.com/en/discussions/managing-discussions-for-your-community/moderating-discussions#marking-a-comment-as-an-answer) so it doesn't show up in the discussion tab as unanswered?

It looks like that's been done by @gsmet. Is there something else for me to click on that I don't see?

**Reply by codespearhead** [reply]
No, it's fine now that an answer has been selected. Thanks!
